### ISSUE TITLE
unused parameter 'file' in tools/objtool/check.c

### ISSUE THREAD

**Issue by tpgxyz** [ClangBuiltLinux/linux#676]
Trying to compile kernel 5.3.1 with LLVM/clang-9.0.0 on x86_64
  
```
HOSTCC  scripts/recordmcount
  CC       /builddir/build/BUILD/linux-5.3/tools/objtool/exec-cmd.o
  CC       /builddir/build/BUILD/linux-5.3/tools/objtool/orc_dump.o
  CC       /builddir/build/BUILD/linux-5.3/tools/objtool/elf.o
  CC       /builddir/build/BUILD/linux-5.3/tools/objtool/special.o
BUILDSTDERR: check.c:807:49: error: unused parameter 'file' [-Werror,-Wunused-parameter]
BUILDSTDERR: static int handle_jump_alt(struct objtool_file *file,
BUILDSTDERR:                                                 ^
```

**Comment by nathanchance** [comment]
While this and #677 do appear to be legitimate warnings, I have never seen them before, leading me to believe this is environment related. I wonder if this patch would hide the warnings for you? https://git.kernel.org/tip/f73b3cc39c84220e6dccd463b5c8279b03514646

**Comment by nathanchance** [comment]
See this thread for more information: https://lore.kernel.org/lkml/133250.1566965715@turing-police/

**Comment by tpgxyz** [comment]
Hi @nathanchance ,
Looks like that patch https://git.kernel.org/tip/f73b3cc39c84220e6dccd463b5c8279b03514646 helps to move further.

**Comment by nathanchance** [comment]
I have requested that that patch be added to -stable: https://lore.kernel.org/stable/20190925164047.GA471117@archlinux-threadripper/

**Comment by nathanchance** [comment]
This is now fixed in 5.3.2: https://git.kernel.org/stable/c/6cc24e4fea57f2aed486a5d9feaacd0a3b1468a3
